pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -400,15 +401,24 @@ func (b *AutoscalingGroupModelBuilder) buildAutoScalingGroupTask(c *fi.ModelBuil
 		}
 
 		if extLB.TargetGroupARN != nil {
+			parsed, err := arn.Parse(fi.StringValue(extLB.TargetGroupARN))
+			if err != nil {
+				return nil, fmt.Errorf("error parsing target grup ARN: %v", err)
+			}
+			resource := strings.Split(parsed.Resource, "/")
+			if len(resource) != 3 || resource[0] != "targetgroup" {
+				return nil, fmt.Errorf("error parsing target grup ARN resource: %q", parsed.Resource)
+			}
 			tg := &awstasks.TargetGroup{
-				Name:   fi.String(fmt.Sprintf("external-tg-%d", i)),
+				Name:   fi.String(resource[1]),
```

Comment:
This assumes that the ARN field after `targetgroup/` is always equal to the target group's name and that AWS never modifies that in anyway. is that a safe assumption?

Reply:
I think for now should be ok. AWS will break pretty much anything it wants :).

Reply:
For now, in case of TGs, the resource ID is always `tg-name/big-number`. This PR validates this format so people will know in advance if something is changed in the future.

Technically, tasks are compared by name and ARN. The name is also used for ordering. We could potentially add some logic that detects the external TGs and generates the name is a way that doesn't assume `tg-name/big-number`, but would complicate things and add assume some fixed names as belonging to kOps. Seems safer to just assume AWS will not change the ARN format for TGs. For sure AWS will not change the ARNs of existing TGs.